Simplenote/Classes/SPTagsListViewController.m

```diff
@@ -483,48 +530,41 @@ - (void)removeTagAtIndexPath:(NSIndexPath *)indexPath {
         [appDelegate.noteListViewController update];
     }
     
-    BOOL lastTag = [self numTags] == 1;
+    BOOL lastTag = [self numberOfTags] == 1;
 
     if ([[SPObjectManager sharedManager] removeTag:tag] && !lastTag) {
-        
         [self.tableView beginUpdates];
-            [self.tableView deleteRowsAtIndexPaths:@[indexPath]
-                                  withRowAnimation:UITableViewRowAnimationLeft];
+        [self.tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationLeft];
         [self.tableView endUpdates];
     } else {
         [self.tableView reloadData];
     }
 }
 
 - (void)renameTagAction:(Tag *)tag {
-    
     if (_renameTag) {
-        
-        SPTagListViewCell *cell = (SPTagListViewCell *)[self.tableView cellForRowAtIndexPath:[NSIndexPath indexPathForItem:[self rowForTag:_renameTag]
-                                                                                                                 inSection:kSectionTags]];
+        SPTagListViewCell *cell = [self cellForTag:_renameTag];
         [cell.tagNameTextField endEditing:YES];
     }
     
     _renameTag = tag;
     
     // begin editing the text field
     SPTagListViewCell *cell = [self cellForTag:tag];
-    
-    [cell setTextFieldEditable:YES];
+
+    cell.tagNameTextField.enabled = YES;
     [cell.tagNameTextField becomeFirstResponder];
 }
 
 
 #pragma mark - UITextFieldDelegate methods
 
 - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string {
-    
     BOOL endEditing = NO;
     if ([string hasPrefix:@" "]) {
         string = nil;
```

Comment:
The static analyzer is complaining about this line `nil assigned to a pointer which is expected to have non-null value`

Reply:
A couple of other places too, but those files weren't modified in this PR :)

Reply:
Since we're here!!! Patching the `SPTagListViewController`'s static analyzer. 

I feel the pull to... rewrite this entire thing!!!! Thank you Eric!